modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/connections/HostConnections.java

```diff
@@ -167,6 +174,24 @@ public void release(NHttpClientConnection conn) {
         }
     }
 
+    private long getExpiryTime(NHttpClientConnection connection) {
+        long expiryTime = System.currentTimeMillis();
+
+        String keepAliveTimeout = (String) connection.getContext().
+                getAttribute(PassThroughConstants.CONNECTION_KEEP_ALIVE_TIME_OUT);
+        if (keepAliveTimeout != null) {
+            try {
+                expiryTime = expiryTime + (Integer.parseInt(keepAliveTimeout) * 1000) - this.connectionGraceTime;
+            } catch (NumberFormatException e) {
+                log.error("Error parsing value " + keepAliveTimeout + " as connection keep-alive timeout.", e);
+                expiryTime = expiryTime + this.connectionIdleTime;
+            }
+        } else {
+            expiryTime = expiryTime + this.connectionIdleTime;
+        }
+        return expiryTime;
```

Comment:
shall we add a debug here log for the expiry time set

Reply:
Fixed